src/components/ExportDialog.tsx

```diff
@@ -71,9 +71,31 @@ const ExportModal = ({
       scale,
       shouldAddWatermark,
     });
-    previewNode?.appendChild(canvas);
+
+    let div: HTMLDivElement | undefined;
+    if (canvas) {
+      previewNode?.appendChild(canvas);
+    } else {
+      const h3 = document.createElement("h3");
+      const p = document.createElement("p");
+      div = document.createElement("div");
+      h3.innerText = "Canvas too big for PNG export";
+      // h3.innerText = t("alerts.CanvasTooBigForPNGExport");
```

Comment:
use it or remove it.

Reply:
Using `innerHTML` or `dangerouslySetInnerHTML` didn't feel right to me, so I created a new React component: `ErrorCanvasPreview`.